sgqlc/endpoint/requests.py

```diff
@@ -109,7 +109,13 @@ def __init__(self, url, base_headers=None, timeout=None, method='POST',
         self.timeout = timeout
         self.method = method
         self.auth = auth
-        self.session = session
+        self.session = session or requests.Session()
+        self.close_session = session is not None
```

Comment:
isn't this flipped? You want to close if `or requests.Session()`, in this case `session is None`

Reply:
My bad